tests/bal_functional_behaviour_suite.py

```diff
@@ -142,6 +144,54 @@ def test_when_getWithRelationshipsPaged_called_then_results_delayed_by_specified
             self.createTestContext(access=Context.Access.kRead),
         )
 
+    def test_when_getWithRelationshipPaged_pager_hasNext_called_then_results_delayed(
+        self,
+    ):
+        for query_latency in self.__simulated_latency_subtests():
+            pager = self.__getWithRelationshipPaged_pager()
+            with self.__assert_simulated_latency_applied(query_latency):
+                pager.hasNext()
+
+    def test_when_getWithRelationshipsPaged_pager_hasNext_called_then_results_delayed(
+        self,
+    ):
+        for query_latency in self.__simulated_latency_subtests():
```

Comment:
[Question] Is it not possible to use `__check_simulated_latencies_without_callbacks` for these methods?

Reply:
The problem IIRC was that there's a "setup" step to get the pager (`getWithRelationshipsPaged` call), and we want to ensure it isn't delayed and doesn't pollute the test of the pager.

Reply:
:+1: